File: Challenges/Coding World Cup/2023/Day 14 - NZ vs AFG - Perfect lineup.java
import java.util.Arrays;

public class Solution {
    static int goodSubarray(int[] a) {
        int n = a.length;
        int maxLength = 1;

        Arrays.sort(a);

        int start = 0;
        int end = 1;
        while (end < n) {
            if (a[end] - a[start] <= 10) {
                maxLength = Math.max(maxLength, end - start + 1);
                end++;
            } else {
                start++;
            }
        }

        return maxLength;
    }
}

File: Challenges/Coding World Cup/2023/Day 14 - NZ vs AFG - Perfect lineup.cpp
int goodSubarray(vector<int> &a)
{
    int n = a.size();
    int maxLength = 1;

    std::sort(a.begin(),
              a.end());

    int start = 0;
    int end = 1;
    while (end < n)
    {
        if (a[end] - a[start] <= 10)
        {
            maxLength = std::max(maxLength, end - start + 1);
            end++;
        }
        else
        {
            start++;
        }
    }

    return maxLength;
}
